### PR TITLE
feat(core): Edit Page ask to copy non-existent language page #27281

### LINKED ISSUE #27281: Edit Page: Resolve Error When Editing Non-Existent Language Page
### Problem Statement
When a user is editing a page in language 2 and this page has no language 1 version, if the user tries to select language 1, the page editor throws an error and kicks the user out of the page editor.

### Steps to Reproduce
https://github.com/dotCMS/core/assets/751424/a4e4c7b1-798d-4e0d-8add-6e2b5f1b2758

```[tasklist]
### Acceptance Criteria
- [ ] User should be able to create the language version if doesn't exist and then edit the page
```

### External Links
New endpoint: https://github.com/dotCMS/core/issues/27185

This endpoint will tell you if the page have or not a version for the selected language, which allows you to change the flow.

So now instead of using the language API, you can use the page language API.

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Confirm to auto-fill innexistent language with the current language on Page

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info

https://github.com/dotCMS/core/assets/1909643/4e43fd16-3cad-44a0-b601-c3be2281ad3a

### CODE REVIEW

**Review by fmontes — COMMENTED** [review]
Looking good.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.spec.ts
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.spec.ts:1]: https://github.com/dotCMS/core/assets/751424/4bc3c9d3-eb07-4008-b22e-08ad312eac71

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.spec.ts:1]: The confirmation dialog:

Title: Create New Language Version?

Message:
Page does not exist in the selected language. Create a new version in [selected language]?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.spec.ts
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.spec.ts:1]: Screenshot 2024-01-12 at 11 09 06 AM

Done.

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.spec.ts:1]: Push push :D

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**